### PR TITLE
Two small fixes in the sfPropelPlugin for newer PHP versions

### PR DESCRIPTION
**Pull request by leonrenkema**
I'v patched the sfPropelPlugin for newer versions of PHP that have deprecated mysql_escape_string

### CODE REVIEW

File: lib/plugins/sfPropelPlugin/lib/vendor/phing/system/io/UnixFileSystem.php [hunk 1]
```diff
@@ -191,7 +191,7 @@ function resolveFile(PhingFile $f) {
     /* -- most of the following is mapped to the php natives wrapped by FileSystem */    
 
     /* -- Attribute accessors -- */
-    function getBooleanAttributes(&$f) {
+    function getBooleanAttributes($f) {
```
Comment (linaori) [inline lib/plugins/sfPropelPlugin/lib/vendor/phing/system/io/UnixFileSystem.php:194]: Is this related to the issue causing output when generating models?

Comment (leonrenkema) [inline lib/plugins/sfPropelPlugin/lib/vendor/phing/system/io/UnixFileSystem.php:194]: This has to do with this strict warning: PHP Strict standards:  Declaration of UnixFileSystem::getBooleanAttributes() should be compatible with FileSystem::getBooleanAttributes($f) in code/lib/vendor/symfony/symfony1/lib/plugins/sfPropelPlugin/lib/vendor/phing/system/io/UnixFileSystem.php on line 278 during generation of the models yes.

File: lib/plugins/sfPropelPlugin/lib/vendor/propel-generator/classes/propel/engine/platform/MysqlPlatform.php [hunk 2]
```diff
@@ -94,8 +94,8 @@ public function hasSize($sqlType)
 	 */
 	public function disconnectedEscapeText($text)
 	{
-		if (function_exists('mysql_escape_string')) {
-			return mysql_escape_string($text);
+		if (function_exists('mysql_real_escape_string')) {
+			return mysql_real_escape_string($text);
```
Comment (linaori) [inline lib/plugins/sfPropelPlugin/lib/vendor/propel-generator/classes/propel/engine/platform/MysqlPlatform.php:98]: Both functions are deprecated as of 5.5. Is this change really needed, or do you need this because of `mysql_set_charset()`?

http://php.net/manual/en/function.mysql-real-escape-string.php

> Warning
> This extension is deprecated as of PHP 5.5.0, and will be removed in the future. Instead, the MySQLi or PDO_MySQL extension should be used. See also MySQL: choosing an API guide and related FAQ for more information. Alternatives to this function include:
> -  mysqli_real_escape_string()
> -  PDO::quote()

Comment (leonrenkema) [inline lib/plugins/sfPropelPlugin/lib/vendor/propel-generator/classes/propel/engine/platform/MysqlPlatform.php:98]: For now, my php 5.4 stopped complaining with this change. But indeed from 5.5+ we have another challenge and it should be ported to mysqli or PDO.

Comment (linaori) [inline lib/plugins/sfPropelPlugin/lib/vendor/propel-generator/classes/propel/engine/platform/MysqlPlatform.php:98]: Besides of an out of memory issue and a preg_match error (most likely a model issue), I get no strict warnings running `./symfony propel:build-model` on 5.5. For me, that method is never executed but the platform does get initialized. 

Which command do you run to trigger this?

Comment (leonrenkema) [inline lib/plugins/sfPropelPlugin/lib/vendor/propel-generator/classes/propel/engine/platform/MysqlPlatform.php:98]: I get it only when building the sql ./symfony propel:build-sql sorry did not mention that earlier.

Comment (linaori) [inline lib/plugins/sfPropelPlugin/lib/vendor/propel-generator/classes/propel/engine/platform/MysqlPlatform.php:98]: I've ran it with the change you have applied on php 5.5, this is the result:

```
[PHP Error] mysql_real_escape_string(): The mysql extension is deprecated and will be removed in the future: use mysqli or PDO instead [line 98 of /vendor/symfony/symfony1/lib/plugins/sfPropelPlugin/lib/vendor/propel-generator/classes/propel/engine/platform/MysqlPlatform.php]
```

**Comment by linaori** [comment]
@leonrenkema Thanks for the PR, I'll release a tag shortly.
